sky/backends/cloud_vm_ray_backend.py

```diff
@@ -591,25 +611,36 @@ def add_epilogue(self) -> None:
 
         self._code += [
             textwrap.dedent(f"""\
-            returncodes = ray.get(futures)
+            returncodes = wait_or_fail(futures)
+            # Remove the placement group after all tasks are done, so that the
+            # next job can be scheduled on the released resources immediately.
+            ray_util.remove_placement_group(pg)
+            sys.stdout.flush()
+            sys.stderr.flush()
             if sum(returncodes) != 0:
                 job_lib.set_status({self.job_id!r}, job_lib.JobStatus.FAILED)
-                # This waits for all streaming logs to finish.
+                # Schedule the next pending job immediately to make the job
+                # scheduling more efficient.
                 job_lib.scheduler.schedule_step()
+                # This waits for all streaming logs to finish.
                 time.sleep(0.5)
+                reason = ''
+                if any(r == 139 for r in returncodes):
+                    reason = '(likely due to Segmentation Fault)'
```

Comment:
This feels like a special-case hack. Maybe it's fine to leave it out?

Reply:
This can be quite useful for the case when people install `pytorch 2.2` in our VM. The program fails with segmentation fault will be really hard to find out what the problem is, as the output will not be streamed. Wdyt?

Reply:
Maybe add 139 = is usually SIGSEGV in comment.